solarnet/roles/ipfs_gateway/templates/nginx_ipfs_gateway.conf.j2

```diff
@@ -76,15 +76,22 @@ server {
 
     resolver 8.8.8.8 8.8.4.4;
 
-    set $hosturi $host$uri;
-    set $proxyhost "";
-    if ($hosturi !~ "^(h\.)?ipfs\.io/(ipfs|ipns|api)(/|$)") {
-        set $proxyhost $host;
+    location /refs/ {
+        rewrite "^/refs/([^/]+)/([^/]+)/([^/]+)$" "/" break;
+        proxy_set_header Host $3.$2.$1.refs.ipfs.io;
```

Comment:
this doesn't seem robust enough to also handle:

```
refs.ipfs.io
c.refs.ipfs.io
b.c.refs.ipfs.io
```

maybe we should just have one ipfs graph, with its root at `refs.ipfs.io`, redirect `/refs/...` to `refs.ipfs.io`, and it could handle `$refs/c/b/a`.